### PR TITLE
change bifunctor documentation; add examples

### PR DESCRIPTION
**Pull request by GinBaby**
I thought the `Bifunctor` documentation could use a little expansion and addition of some examples.

### CODE REVIEW

File: libraries/base/Data/Bifunctor.hs [hunk 1]
```diff
@@ -59,22 +67,49 @@ class Bifunctor p where
...
+    -- Left 'J'
     first :: (a -> b) -> p a c -> p b c
     first f = bimap f id
 
+
     -- | Map covariantly over the second argument.
     --
     -- @'second' ≡ 'bimap' 'id'@
+    --
+    --==== Examples:
```
Comment (bgamari) [inline libraries/base/Data/Bifunctor.hs]: Do you know whether a space is required here? I'm a bit worried that haddock will get confused here.

Comment (GinBaby) [reply libraries/base/Data/Bifunctor.hs]: I also do not know whether it's required, but I've fixed my PR to match the other examples of it I could find in `base` (e.g., in Data.List). Sorry for confusion.

**Comment by bgamari** [comment]
Thanks!

**Comment by bgamari** [comment]
Merged with 275ac8ef0a0081f16abbfb8934e10cf271573768.

**Pull request closed without merging**